src/openklant/components/klantinteracties/api/tests/test_klantcontacten.py

```diff
@@ -1652,19 +1659,213 @@ def test_filter_onderwerpobject_by_klantcontact_filters(self):
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.assertFalse(Betrokkene.objects.filter(uuid=betrokkene.uuid).exists())
+
+        self.assertFalse(DigitaalAdres.objects.filter(uuid=digitaaladres.uuid).exists())
+
+    def test_digitaaladres_not_deleted_if_linked_to_partij(self):
+        klantcontact = KlantcontactFactory.create()
+        betrokkene = BetrokkeneFactory.create(klantcontact=klantcontact)
+
+        partij = PartijFactory.create()
+        digitaaladres = DigitaalAdresFactory.create(
+            betrokkene=betrokkene, partij=partij
+        )
+
+        onderwerpobject = OnderwerpobjectFactory.create(klantcontact=klantcontact)
+
+        url = reverse(
+            "klantinteracties:onderwerpobject-detail",
+            kwargs={"uuid": onderwerpobject.uuid},
+        )
+
+        response = self.client.delete(f"{url}?cascade=true")
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.assertFalse(Betrokkene.objects.filter(uuid=betrokkene.uuid).exists())
+        self.assertTrue(DigitaalAdres.objects.filter(uuid=digitaaladres.uuid).exists())
+
+    def test_destroy_cascade_deletes_bijlage(self):
+        klantcontact = KlantcontactFactory.create()
+        onderwerpobject = OnderwerpobjectFactory.create(klantcontact=klantcontact)
+        bijlage = BijlageFactory.create(klantcontact=klantcontact)
+
+        url = reverse(
+            "klantinteracties:onderwerpobject-detail",
+            kwargs={"uuid": onderwerpobject.uuid},
+        )
+        response = self.client.delete(f"{url}?cascade=true")
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.assertFalse(
+            Onderwerpobject.objects.filter(uuid=onderwerpobject.uuid).exists()
+        )
+        self.assertFalse(Klantcontact.objects.filter(uuid=klantcontact.uuid).exists())
+        self.assertFalse(Bijlage.objects.filter(uuid=bijlage.uuid).exists())
+
+    def test_destroy_cascade_deletes_internetaak(self):
+        klantcontact = KlantcontactFactory.create()
+        onderwerpobject = OnderwerpobjectFactory.create(klantcontact=klantcontact)
+        interne_taak = InterneTaakFactory.create(
+            klantcontact=klantcontact, status=Taakstatus.te_verwerken
+        )
+
+        url = reverse(
+            "klantinteracties:onderwerpobject-detail",
+            kwargs={"uuid": onderwerpobject.uuid},
+        )
+        response = self.client.delete(f"{url}?cascade=true")
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.assertFalse(
+            Onderwerpobject.objects.filter(uuid=onderwerpobject.uuid).exists()
+        )
+        self.assertFalse(Klantcontact.objects.filter(uuid=klantcontact.uuid).exists())
+        self.assertFalse(InterneTaak.objects.filter(uuid=interne_taak.uuid).exists())
+
+    def test_destroy_cascade_deletes_actors(self):
```

Comment:
Because the Actor itself is not deleted, which seems correct imo, because I don't think we want to remove the municipality employees from Open Klant if a Klantcontact related to a Zaak is deleted
```suggestion
    def test_destroy_cascade_deletes_actorklantcontact(self):
```